bindings/rust/standard/integration/src/capability_check.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{future::Future, panic::AssertUnwindSafe};
+
+/// The libcrypto that s2n-tls is linked against.
+#[derive(Debug, PartialEq, Eq)]
+enum Libcrypto {
+    Awslc,
+    AwslcFips,
+    OpenSsl102,
+    OpenSsl111,
+    OpenSsl30,
+}
+
+impl Libcrypto {
+    /// Retrieve the libcrypto from the `S2N_LIBCRYPTO` env variable if available,
+    /// otherwise return "awslc".
+    ///
+    /// S2N_LIBCRYPTO is set in CI as well as the Nix devshell.
+    fn from_env() -> Self {
+        let libcrypto = match std::env::var("S2N_LIBCRYPTO") {
+            Ok(libcrypto) => libcrypto,
+            Err(_) => "awslc".to_string(),
+        };
```

Comment:
Do we have any way to log / message that this default happened? I can see this leading to tricky to debug issues :sweat_smile: